src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -71,10 +75,23 @@ export const useAnalysisFiles = (): AnalysisFileStore => {
       await Ajax(signal).AzureStorage.listNotebooks(workspaceInfo.workspaceId)
     setAnalyses(existingAnalyses)
   }) as () => Promise<void>
+
+  const create: (fullAnalysisName: any, toolLabel: ToolLabel, contents: any) => Promise<void> = _.flow(
+    // @ts-expect-error
+    withErrorReporting('Error creating analysis files'),
+    Utils.withBusyState(setPendingCreate)
+  )(async (fullAnalysisName: any, toolLabel: ToolLabel, contents: any): Promise<void> => {
+    const workspaceInfo = workspace.workspace
+    isGoogleWorkspaceInfo(workspaceInfo) ?
+      await Ajax().Buckets.analysis(workspaceInfo.googleProject, workspaceInfo.bucketName, fullAnalysisName, toolLabel).create(contents) :
+      await Ajax().AzureStorage.blob(workspaceInfo.workspaceId, fullAnalysisName).create(contents)
+    refresh()
+  }) as () => Promise<void>
+
   useEffect(() => {
     refresh()
   }, [workspace]) // eslint-disable-line react-hooks/exhaustive-deps
-  return { refresh, loadedState: { status: loading ? 'Loading' : 'Ready', state: analyses } }
+  return { refresh, create, loadedState: { status: loading ? 'Loading' : 'Ready', state: analyses }, pendingCreate }
```

Comment:
loadedState is only expressing 'Loading' and 'Ready' cases here.  Consider leveraging the new useLoadedData helper hook to express 'Error' cases as well.